components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx

```diff
@@ -0,0 +1,305 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import createWidget from '../widget/index'
+import * as Styled from './style'
+import { StyledTitleTab } from '../widgetTitleTab'
+import assetIcons from './assets/icons'
+import fiatData from './fiat'
+import BitcoinDotComLogo from './assets/logo.png'
+import { CaratDownIcon } from 'brave-ui/components/icons'
+import { getLocale } from '../../../../common/locale'
+
+interface State {
+  selectedAsset: string
+  selectedFiat: string
+  currentAmount: string
+  assetsShowing: boolean
+  fiatCurrenciesShowing: boolean
+}
+
+interface Props {
+  showContent: boolean
+  stackPosition: number
+  onShowContent: () => void
+  onBuyCrypto: () => void
+  onInteraction: () => void
+}
+
+class BitcoinDotCom extends React.PureComponent<Props, State> {
+  private assets: Record<string, string>
+  private fiatCurrencies: Record<string, string>
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      selectedAsset: 'BCH',
+      selectedFiat: 'EUR',
+      currentAmount: '',
+      assetsShowing: false,
+      fiatCurrenciesShowing: false
+    }
+    this.assets = {
+      'BCH': 'Bitcoin Cash',
+      'BTC': 'Bitcoin',
+      'ETH': 'Ethereum',
+      'LTC': 'Litecoin',
+      'XLM': 'Stellar',
+      'XRP': 'Ripple'
+    }
+    this.fiatCurrencies = fiatData
+  }
+
+  renderTitle () {
+    const { showContent } = this.props
+
+    return (
+      <Styled.Header>
+        <Styled.Title contentShowing={showContent}>
+          <Styled.BitcoinDotComIcon>
+            <img src={BitcoinDotComLogo} />
+          </Styled.BitcoinDotComIcon>
+          <Styled.TitleText>
+            {'Bitcoin.com'}
+          </Styled.TitleText>
+        </Styled.Title>
+      </Styled.Header>
+    )
+  }
+
+  toggleAssetsShowing = () => {
+    this.setState({
+      assetsShowing: !this.state.assetsShowing
+    })
+    this.props.onInteraction()
+  }
+
+  setSelectedAsset = (asset: string) => {
+    this.setState({
+      assetsShowing: false,
+      selectedAsset: asset
+    })
+  }
+
+  setCurrentAmount = ({ target }: any) => {
+    this.setState({
```

Comment:
We're not validating this input and forcing the user to use a specific format. Should we be?

Reply:
Added simple validation, should be numerical, parseable floats allowed for fractional fiat, Ex: (200 and 200.50) supported